### PR TITLE
166075655 population dashboards to pdf

### PR DESCRIPTION
**Pull request by jgoley**

### CODE REVIEW

**Comment by jgoley** [comment]
@eanders I brought in [Grover](https://github.com/Studiosity/grover) and [Puppeteer](https://github.com/GoogleChrome/puppeteer).
The PDFs are exporting okay with the charts but there appears to be some bugginess in using `page-break` css. I've wrapped several containers and added a `page-break` div with minimal luck. This is a known issue: https://nathanfriend.io/2019/04/15/pdf-gotchas-with-headless-chrome.html
I'll dig further to see if there's some style on the containers that's preventing break.

I added a button to the bottom of the page which allows downloading the reports as a PDF.

Screen Shot 2019-07-12 at 12 33 41 PM

**Comment by eanders** [comment]
That's looking great.  Is there any way to put in a page-break-avoid?  That might be all we need for a first pass.

**Comment by jgoley** [comment]
I'll double check but I think added `page-break-inside: avoid` to `.ho-container`. There must be something within that's throwing it off. I'll take a closer look.
Oh, also: the headless Chromium window has trouble loading relatively linked assets. Currently I'm using the wicked tag helpers which gather the styles/JS and dump them inline. I'll also worked to fix this so we aren't also dependent on wicked.

**Comment by jgoley** [comment]
I thought about the page-break issue a little more and it may be that the body is not rendering at the correct height or a flex element's height is being calculated incorrectly. I'll investigate those possible issues.

File: config/initializers/npm.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+system 'npm install' if Rails.env.development? || Rails.env.test?
```
Comment (eanders) [inline config/initializers/npm.rb]: Do we need to do something special for production?

**Comment by eanders** [comment]
One, hopefully small request.  Can we put a logo across from the title (where the blue box is in this screenshot?.)  Since they'll be printing these out, I think it makes sense to have some attribution.
Here's an example of what we've done in the past. 
https://github.com/greenriver/hmis-warehouse/blob/master/app/views/window/clients/history/pdf.haml#L5

Screen Shot 2019-07-15 at 10 17 30 AM

File: config/initializers/npm.rb (hunk 1 above)
Comment (jgoley) [reply config/initializers/npm.rb]: Good point we should change that to production if it looks like the correct way of running the command. I think this was on a guide for an app that's deployed to a heroku server. I suppose we can manage the packages manually in the development env?

**Comment by jgoley** [comment]
Yes to the logo request, Puppeteer supports headers and footers although I had trouble getting them to render when testing. Any chance the logos could be inlined asSVG? Otherwise we may need to output them as Base64.

**Pull request merged**